src/components/Editor/Tabs.js

```diff
@@ -484,16 +484,20 @@ class SourceTabs extends PureComponent<Props, State> {
 
 export default connect(
   state => {
-    const selectedSource = getSelectedSource(state);
-    const sourceId = selectedSource ? selectedSource.get("id") : "";
+    const sourceTabs = getSourcesForTabs(state);
+    const sourceTabsMetaData = {};
+    sourceTabs.forEach(source => {
+      const sourceId = source ? source.get("id") : "";
```

Comment:
im wondering, what is the case when a source does not have an id? do you have an example?

also, meta thought: it feels like the meta data belongs to the tab as a data structure too..

Reply:
This was copied from the latest code, so I'm not entirely sure:
```diff
- const selectedSource = getSelectedSource(state);
- const sourceId = selectedSource ? selectedSource.get("id") : "";
+ sourceTabs.forEach(source => {
+   const sourceId = source ? source.get("id") : "";
```
Is there a case where source doesn't have an id?

----
A funny thing about metadata btw, it's set on "LOAD_SOURCE_TEXT", which means that in order for `isReactComponent` to be set tab needs to be visited at least once (if you open a few tabs and reload the window, logos won't show up until you click on respective tabs). It doesn't affect UX that much, and when I tried moving it to reducer it disintegrated my CPU, so I probably didn't entirely understand how it works outside of Tabs component, so I didn't touch it.

Reply:
@fivepointseven that's a good observation. We should probably re-think source metadata so it is a field in sources and tabs, so that when we re-show a tab on reload it is saved...

that way you don't need to revisit it.

Reply:
@jasonLaster What's the starting point in source's "life"? Cause I tried adding that at `ADD_SOURCES` action but it ended up eating all of my resources over 10 seconds after loading so I probably did it wrong.

Reply:
haha - yeah. we don't want to parse every source! it would kill us :)

don't worry about that bug now, I'll add some context in another issue

Reply:
Okay, cool then. So, do you have any issues with this PR I should fix? :)